pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -505,11 +506,51 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		}
 	}
 
+	clusterSync := &hiveintv1alpha1.ClusterSync{}
+	err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Name}, clusterSync)
+	if err != nil {
+		var changed bool
+		var conds []hivev1.ClusterDeploymentCondition
+		if apierrors.IsNotFound(err) {
+			if paused, err := strconv.ParseBool(cd.Annotations[constants.SyncsetPauseAnnotation]); err == nil && paused {
+				cdLog.Info("SyncSet is paused. ClusterSync will not be created")
+				conds, changed = controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+					cd.Status.Conditions,
+					hivev1.SyncSetFailedCondition,
+					corev1.ConditionTrue,
+					"SyncSetPaused",
+					"SyncSet is paused. ClusterSync will not be created",
+					controllerutils.UpdateConditionIfReasonOrMessageChange,
+				)
+			} else {
+				cdLog.Info("ClusterSync has not yet been created")
+				conds, changed = controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+					cd.Status.Conditions,
+					hivev1.SyncSetFailedCondition,
+					corev1.ConditionTrue,
+					"MissingClusterSync",
+					"ClusterSync has not yet been created",
+					controllerutils.UpdateConditionIfReasonOrMessageChange,
+				)
+			}
+			// In case the clusterSync has been deleted at this point, clear the clusterSyncFailing metric
+			clustersync.ClearClusterSyncFailingSecondsMetric(clusterSync, cdLog)
+		} else {
+			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not get ClusterSync")
```

Comment:
By not erroring here, we're passing an empty clustersync around. Hopefully the worst that will happen is we'll clear a metric with an empty label, which should be a no-op, but it's not future-proof.